### DISCUSSION TITLE
Can `prost_build` generate 1 module per proto?

### DISCUSSION THREAD

**Discussion by bitwizeshift** [tokio-rs/prost#755] [Q&A]
I'm new to using `prost` and `prost_build`, so it's likely that I'm just overlooking this feature; but is it possible to have a 1:1 mapping between `*.proto` files and the generated `*.rs` files, rather than generating everything into a file that corresponds to the package identifier?

I'm working with a relatively large protobuf project with over a hundred semantic-type definitions used for messaging. All of these types exist in the same package definition, and the definitions of which are _not within my control_. `prost` by default is generating all of these into a single `rs` file, which is nearly **68k lines of code**. This affects indexing greatly, massively increases compile-times (25+ seconds), and breaks IDEs that have lower file-size thresholds for intellisense.

Is there a setting I could do that allows this to be broken up, so that rather than generating one `my.package.id.rs`, it generates something like:

```
my.package.id/
|- foo.rs
|- bar.rs
|- ... etc
```

Is this possible?

**Comment by LucioFranco** [comment]
Prost doesn't support this though this could be something we add as an opt-in behavior.
